cloudbuild.yaml

```diff
@@ -0,0 +1,24 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+
+# this must be specified in seconds. If omitted, defaults to 600s (10 mins)
+timeout: 1200s
+# this prevents errors if you don't use both _GIT_TAG and _PULL_BASE_REF,
+# or any new substitutions added in the future.
+options:
+  substitution_option: ALLOW_LOOSE
+steps:
+  - name: 'gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4'
+    entrypoint: make
+    env:
+    - DOCKER_CLI_EXPERIMENTAL=enabled
+    - VERSION=$_GIT_TAG
+    - BASE_REF=$_PULL_BASE_REF
+    args:
+    - push
+substitutions:
+  # _GIT_TAG will be filled with a git-based tag for the image, of the form vYYYYMMDD-hash, and
+  # can be used as a substitution
+  _GIT_TAG: '12345'
```

Comment:
What does this mean? Will this be substituted by the git-tag `v20190906-745fed4` ?

Reply:
Short answer ...

Yeah it looks like it will be vYYYYMMDD-GIT_TAG. For example v20200131-v0.10.0. I think this is not what we want. When a descheduler version is released the tag should be something like v0.10.0. I'll have to figure out how to change this.

Long answer ...

It looks like this a [GCB user defined substitution](https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values#using_user-defined_substitutions). The container image listed on line 10 must be going something special with the variable.

I believe I tracked it down to this directory https://github.com/kubernetes/test-infra/tree/master/images/builder. More specifically here ... https://github.com/kubernetes/test-infra/blob/09a81c4adf5b662b2643dccb9d285ad3102af4af/images/builder/main.go#L48-L56.

It seems like most other k8s projects using image build automation are using _GIT_TAG. See [here](https://github.com/kubernetes/kops/blob/master/ci/postsubmit/push-to-staging/cloudbuild.yaml#L10), [here](https://github.com/kubernetes-sigs/external-dns/blob/master/cloudbuild.yaml#L10) and [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/cloudbuild.yaml#L10).

Reply:
One more thought ...

We could merge this PR as is and sort out git/docker tag configuration is a follow up PR. This would at least enable the automated builds for the master branch and let us confirm that the build automation is working.

Also be aware that the container build automation only pushes container images the GCR staging registry. A person will still need to submit a pull request for each container image that needs to be promoted to the production GCR registry.

Reply:
Another thought (which is where the design comes from): it doesn't really matter what you tag things in the staging registry - when the image promoter promotes them, they can be given a different tag to what they had in staging.

Reply:
@Katharine thanks for adding your comment. This makes sense to me now. We don't need to worry about the tags in the staging registry.

@ravisantoshgudimetla let me know if you have any additional questions on this topic.

Reply:
So, with this as it is we'll get staging repo tags in the format `v20190906-v0.9.0-76-g30d05382b` (based on my own run of the function you linked to [here](https://github.com/kubernetes/test-infra/blob/09a81c4adf5b662b2643dccb9d285ad3102af4af/images/builder/main.go#L48-L56)).

That seems good enough for me in the staging repo, it's descriptive of the version. And we choose our published tag when promoted, so it will be cleaner (like `v0.9.0`). Sgtm